raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -628,7 +629,7 @@ void deliverResponse(int correlationId, Node source, ApiMessage response) {
     RaftRequest.Outbound assertSentBeginQuorumEpochRequest(int epoch, int numBeginEpochRequests) {
         List<RaftRequest.Outbound> requests = collectBeginEpochRequests(epoch);
         assertEquals(numBeginEpochRequests, requests.size());
-        return requests.get(0);
+        return !requests.isEmpty() ? requests.get(0) : null;
```

Comment:
Let's avoid returning `null` from "public" methods. We normally don't test the absence of a side effect because those checks are technically infinite. Maybe we can expose `collectBeginEpochRequests` as package private and check the size of the list.

Reply:
only one test actually needs the request data - I'll keep this helper to minimize redundancy of assertions, and rewrite the one test to grab the request data from collectBeginEpochRequests directly.